doc/api/deprecations.md

```diff
@@ -2712,6 +2711,22 @@ Type: Documentation-only.
 
 Prefer [`message.socket`][] over [`message.connection`][].
 
+### DEP0150: Folder index for "type": "module"
+<!-- YAML
+changes:
+  - version: REPLACEME
+    pr-url: https://github.com/nodejs/node/pull/36918
+    description: Runtime deprecation.
+-->
+
+Type: Documentation (supports [`--pending-deprecation`][])
+
+Previously, `index.js` lookup would apply for `require('pkg')` even for packages
+with a `"type": "module"` field in their package.json
+
+With this deprecation, all packages with `"type": "module"` require an explicit
+`"exports"` field entry point to resolve for the main.
```

Comment:
```suggestion
`"exports"` field.
```

Technically I think just the key is required, not that it also specifies a main entry point (right?). Since packages might supply several coequal entry points.

Reply:
Subpaths remain supported, this just deprecates the `"main"` as the package entry point.

That does seem inconsistent though.